numpyro/contrib/funsor/enum_messenger.py

```diff
@@ -253,9 +253,9 @@ def _get_dim_to_name(batch_shape, dim_to_name=None, dim_type=DimType.LOCAL):
 
         # interpret all names/dims as requests since we only run this function once
         for dim in range(-batch_dim, 0):
-            if batch_shape[dim] == 1:
-                continue
             name = dim_to_name.get(dim, None)
+            if batch_shape[dim] == 1 and name is None:
+                continue
```

Comment:
Per discussion on Slack, this change was made because the time dimension on the left sometimes necessitates empty dimensions appearing before they have been assigned a name.

Reply:
yeah, I'll add this comment. thanks! :D